Husky commit hooks cool but incomplete

```javascript
    "precommit": "npm run lint"
```

This is nice but it doesn't test if the files being linted are actually staged for commit.

I have a problem with this: It aborts commits for files I'm currently not editing. It aborted within an interactive rebase lol:

```
➜  js-bigchaindb-driver git:(http-api-1.0-rc-changes) git rebase -i HEAD^^^^^^ 
Rebasing (1/6)
> husky - npm run -s precommit
> husky - node v8.1.1


/home/tim/ascribe/js-bigchaindb-driver/test/connection/test_connection.js
  3:8  error  Unexpected namespace import  import/no-namespace

✖ 1 problem (1 error, 0 warnings)


> husky - pre-commit hook failed (add --no-verify to bypass)
> husky - to debug, use 'npm run precommit'
Could not amend commit after successfully picking 88f8b36108770328fe5024cdbb176ac2d8324455... Skip test for Connection._req
This is most likely due to an empty commit message, or the pre-commit hook
failed. If the pre-commit hook failed, you may need to resolve the issue before
you are able to reword the commit.
You can amend the commit now, with

        git commit --amend 

Once you are satisfied with your changes, run

        git rebase --continue

➜  js-bigchaindb-driver git:(88f8b36) 
```

--- a/package.json
+++ b/package.json
@@ -28,7 +28,7 @@
     "report-coverage": "nyc report --reporter=lcov > coverage.lcov && codecov"
   },
   "lint-staged": {
-    "*.js": ["npm run lint"]
+    "*.js": ["eslint"]
   },
   "devDependencies": {
     "ava": "^0.19.1",

--- a/package.json
+++ b/package.json
@@ -24,8 +24,11 @@
     "release-minor": "./node_modules/release-it/bin/release.js minor --src.tagName='v%s' --github.release --npm.publish --non-interactive",
     "release-major": "./node_modules/release-it/bin/release.js major --src.tagName='v%s' --github.release --npm.publish --non-interactive",
     "prepublishOnly": "npm update && npm run build",
-    "precommit": "npm run lint",
+    "precommit": "lint-staged",
     "report-coverage": "nyc report --reporter=lcov > coverage.lcov && codecov"
+  },
+  "lint-staged": {
+    "*.js": ["npm run lint"]
   },
   "devDependencies": {
     "ava": "^0.19.1",
@@ -46,6 +49,7 @@
     "eslint-config-ascribe": "^3.0.4",
     "eslint-plugin-import": "^2.2.0",
     "husky": "^0.13.4",
+    "lint-staged": "^4.0.0",
     "nyc": "^11.0.2",
     "release-it": "^2.7.3",
     "rimraf": "^2.5.4",

